### PR TITLE
Adds tests and improved handlers for specific http status codes

### PR DESCRIPTION
**Pull request by Erwinvandervalk**
Slightly better handling for timeouts and situations where the server cannot be found.

### CODE REVIEW

File: src/ProxyKit.Tests/ProxyKit.Tests.csproj [hunk 1]
```diff
@@ -20,4 +21,10 @@
     <ProjectReference Include="..\ProxyKit\ProxyKit.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="Microsoft.AspNetCore.Server.HttpSys">
+      <HintPath>..\..\..\..\Users\erwinv\.nuget\packages\microsoft.aspnetcore.server.httpsys\2.1.1\lib\netstandard2.0\Microsoft.AspNetCore.Server.HttpSys.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (Erwinvandervalk) [reply src/ProxyKit.Tests/ProxyKit.Tests.csproj]: good catch.. thanks!

File: src/ProxyKit.Tests/EndToEndTests.cs [hunk 2]
```diff
@@ -1,62 +1,231 @@
...
+                    // When server is stopped, should return 
+                    await server.StopAsync();
+                    result = await client.GetAsync("/realServer/normal");
+                    if (result != null)
+                    {
+                        try
+                        {
+                            result.StatusCode.ShouldBe(HttpStatusCode.GatewayTimeout);
+                        }
+                        catch (NullReferenceException)
```
Comment (Erwinvandervalk) [inline src/ProxyKit.Tests/EndToEndTests.cs]: @osoykan Strangely enough, this test fails with a null reference exception on the build server.. But I don't see why.. do you have any thoughts?

File: src/ProxyKit.Tests/ProxyKit.Tests.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (damianh) [reply src/ProxyKit.Tests/ProxyKit.Tests.csproj]: Any reason to use HttpSys over Kestrel? (HttpSys won't run on linux / Travis)

[the remark this answers is not in the document]
Comment (Erwinvandervalk) [reply src/ProxyKit.Tests/ProxyKit.Tests.csproj]: I wasn't using httpsys.. just added the reference when I was fidgeting ;) removed it again.

File: src/ProxyKit.Tests/EndToEndTests.cs (hunk 2 above)
Comment (osoykan) [reply src/ProxyKit.Tests/EndToEndTests.cs]: As a result of running the build pipeline in a linux docker container; it shows us the solution will be:
https://github.com/osoykan/ProxyKit/blob/patch-1/src/ProxyKit/HttpContextExtensions.cs#L56-L73

Sent a PR: https://github.com/Erwinvandervalk/ProxyKit/pull/2

**Pull request merged**